mrp_unbuild_cust_alu_analytics/models/mrp_unbuild.py

```diff
@@ -0,0 +1,187 @@
+        folder_path = os.path.join(path)
+        processed_folder_path = os.path.join(path, 'processed')
+        if not os.path.exists(folder_path):
+            try:
+                os.makedirs(folder_path)
+                logger.info(
+                    "Created folder on %s" % (folder_path)
+                )
+            except OSError as e:
+                logger.critical(
+                    "Can't create folder on %s: %s" % (folder_path, e)
+                )
+                raise UserError(_("Can't create folder on %s: %s" % (folder_path, e)))
+
+        if not os.path.exists(processed_folder_path):
+            try:
+                os.makedirs(processed_folder_path)
+                logger.info(
+                    "Created folder on %s" % (processed_folder_path)
+                )
+            except OSError as e:
+                logger.critical(
+                    "Can't create folder on %s: %s" % (processed_folder_path, e)
+                )
+                raise UserError(_("Can't create folder on %s: %s" % (processed_folder_path, e)))
+
+        files = os.listdir(folder_path)
+        files.remove(processed_folder_path.split("/")[-1])
+        if files:
+            for file in files:
+                unbuild_id = self.env['mrp.unbuild']
+                unbuild_name = ""
+                all_data = []
+                if file.endswith('.xml'):
+                    # Get xml data
+                    tree = etree.parse(folder_path + '/' + file)
+                    root = tree.getroot()
+                    data = self.get_element_and_children_data(root)
+                    analytics = []
+                    for attr in data['children'][0]['children'][0]['children']:
+                        if attr['children'][0]['text'] == 'Proceso':
+                            unbuild_id = unbuild_id.search([('name', '=', attr['children'][1]['text'])])
+                            unbuild_name = attr['children'][1]['text']
+                    for analytic in data['children'][0]['children'][2]['children'][0]['children'][1]['children']:
+                        for element in analytic['children']:
+                            if element['attributes']:
+                                if element['attributes']['StatType'] == 'Mean':
+                                    analytics.append({
+                                        analytic['attributes']['ElementName']: float(element['children'][0]['text'])
+                                    })
+
+                    #Process xml data
+                    if unbuild_id:
+                        if unbuild_id.state == 'done':
+                            move_id = False
+                            move_line_ids = unbuild_id.produce_line_ids.move_line_ids
+                            for move_line in move_line_ids.move_id:
+                                if "Virtual Locations" in move_line.location_dest_id.complete_name:
```

Comment:
This is a bad approximation... you mean "the move that has Virtual Locations/Production as destination location", don't you? If I'm right, this is strictly better and not depending on translations, etc:
```suggestion
                                if move_line.location_dest_id.usage == "production":
```

Reply:
Pending to be answered/fixed

Reply:
Solved